src/utiltime.cpp

```diff
@@ -3,26 +3,85 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#define __STDC_WANT_LIB_EXT1__ 1 // for gmtime_s
+
 #if defined(HAVE_CONFIG_H)
 #include "config/bitcoin-config.h"
 #endif
 
 #include "utiltime.h"
 
+#include <chrono>
+#include <thread>
+#include <locale>
+#include <sstream>
+#include <string.h>
+#include <time.h>
 #include <boost/date_time/posix_time/posix_time.hpp>
 #include <boost/thread.hpp>
 
 using namespace std;
 
 static int64_t nMockTime = 0; //!< For unit testing
 
+static inline const tm gmtime_int(time_t time)
+{
+    tm out = {};
+#if defined(HAVE_W32_GMTIME_S)
+    gmtime_s(&out, &time);
+#elif defined(HAVE_C11_GMTIME_S)
+    gmtime_s(&time, &out);
+#elif defined(HAVE_GMTIME_R)
+    gmtime_r(&time, &out);
+#else
+    // Not thread-safe!
+    out = *gmtime(&time);
+#endif
+    return out;
+}
+
+bool ChronoSanityCheck()
```

Comment:
@laanwj Best I can tell from google and reading the c spec, it's not guaranteed to be based on the UNIX epoch either :(. Of course, I've never seen an implementation that isn't, as that would likely break years worth of assumptions. And based on that, I would assume that c++ implementations will be using the underlying c implementation.

So my logic was: if we can run a quick sanity check here to be sure that c/c++/unix epoch are all aligned, then we can use c apis and std::chrono interchangeably throughout the codebase without worry.

As for why chrono here, it was to avoid gettimeofday portability issues. If you'd prefer to work around those instead, no problem.

You're right though that DateTimeStrFormat is kinda clunky. I only used c++ there to avoid c string size guessing. I'm also happy to switch that if you'd prefer.